library/src/main/scala/scala/scalajs/js/Symbol.scala

```diff
@@ -0,0 +1,41 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2017, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+
+/** <span class="badge badge-ecma6" style="float: right;">ECMAScript 6</span>
+ *  JavaScript Symbol.
+ */
+@js.native
+sealed trait Symbol extends js.Any
+
+object Symbol {
+  @JSName("Symbol")
+  @js.native
+  private object NativeSymbol extends js.Object {
+    def apply(): Symbol = js.native
+    def apply(description: String): Symbol = js.native
+
+    def `for`(key: String): Symbol = js.native
+    def keyFor(sym: Symbol): js.UndefOr[String] = js.native
+
+    val iterator: Symbol = js.native
+  }
```

Comment:
Don't you mean turning `scala.Symbol` into a hijacked class? I do get the story for `js.Array`, but `js.Array` is also much closer to the runtime. And I think between 0.6 and 1.0 its ok to have the theoretically source incompatible change of making `js.Symbol` not extend `js.Any` anymore.